tests/test_when.py

```diff
@@ -1835,12 +1815,136 @@ class test_when_alwcomb_order(m.Circuit):
             x @= ~io.I
             io.O @= ~x
 
-    with pytest.raises(MlirWhenCycleError):
-        m.compile(
-            "build/test_when_alwcomb_order",
-            test_when_alwcomb_order,
-            output=_OUTPUT_TYPE
-        )
+    m.compile(
```

Comment:
wondering if all these tests following the same structure could be run using a test harness -- or simpler just a helper function like `run_test()`